CMakeLists.txt

```diff
@@ -204,51 +203,24 @@ if(NOT TARGET uninstall)
     COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/uninstall.cmake)
 endif()
 
-# -- utility targets -----------------------------------------------------------
-
-if(CAF_ENABLE_UTILITY_TARGETS)
-  add_executable(caf-generate-enum-strings
-                 EXCLUDE_FROM_ALL
-                 cmake/caf-generate-enum-strings.cpp)
-  target_link_libraries(caf-generate-enum-strings PRIVATE CAF::internal)
-  add_custom_target(consistency-check)
-  add_custom_target(update-enum-strings)
-  # adds a consistency check that verifies that `cpp_file` is still valid by
-  # re-generating the file and comparing it to the existing file
-  function(caf_add_enum_consistency_check hpp_file cpp_file)
-    set(input "${CMAKE_CURRENT_SOURCE_DIR}/${hpp_file}")
-    set(file_under_test "${CMAKE_CURRENT_SOURCE_DIR}/${cpp_file}")
-    set(output "${CMAKE_CURRENT_BINARY_DIR}/check/${cpp_file}")
-    get_filename_component(output_dir "${output}" DIRECTORY)
-    file(MAKE_DIRECTORY "${output_dir}")
-    add_custom_command(OUTPUT "${output}"
-                       COMMAND caf-generate-enum-strings "${input}" "${output}"
-                       DEPENDS caf-generate-enum-strings "${input}")
-    get_filename_component(target_name "${input}" NAME_WE)
-    add_custom_target("${target_name}"
-                      COMMAND
-                        "${CMAKE_COMMAND}"
-                        "-Dfile_under_test=${file_under_test}"
-                        "-Dgenerated_file=${output}"
-                        -P "${PROJECT_SOURCE_DIR}/cmake/check-consistency.cmake"
-                      DEPENDS "${output}")
-    add_dependencies(consistency-check "${target_name}")
-    add_custom_target("${target_name}-update"
-                      COMMAND
-                        caf-generate-enum-strings
-                        "${input}"
-                        "${file_under_test}"
-                       DEPENDS caf-generate-enum-strings "${input}")
-    add_dependencies(update-enum-strings "${target_name}-update")
-  endfunction()
-else()
-  function(caf_add_enum_consistency_check hpp_file cpp_file)
-    # nop
-  endfunction()
-endif()
-
 # -- utility functions ---------------------------------------------------------
 
+# adds a consistency check that verifies that `cpp_file` is still valid by
+# re-generating the file and comparing it to the existing file
```

Comment:
This is wrong, I think. There is not consistency check. This simply generates the files, right?